redis/docker-entrypoint.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+set -e
+
+# first arg is `-f` or `--some-option`
+# or first arg is `something.conf`
+if [ "${1#-}" != "$1" ] || [ "${1%.conf}" != "$1" ]; then
+	set -- redis-server "$@"
+fi
+
+# allow the container to be started with `--user`
+if [ "$1" = 'redis-server' -a "$(id -u)" = '0' ]; then
+	find . \! -user redis -exec chown redis '{}' +
+	echo "su redis -s /bin/bash -c \"$0 $@\"" | exec
+fi
```

Comment:
I feel a bit confused here but how does this support --user? This just hardcodes running redis as the redis user if it was run without the --user option.

Reply:
It is a translation from  the official redis docker files for the purpose to keep best compatibility for the users.
https://github.com/docker-library/redis/blob/dcc0a2a343ce499b78ca617987e8621e7d31515b/5.0/docker-entrypoint.sh

Yes, your findings is correct, I am a little bit confused as well.
Maybe I delete the comments, what do you think?

Reply:
Well as long as we keep compatibility with the official image this is fine with me. I'd just delete the comment and then I'll merge the change.